tools/build/Makefile.rules

```diff
@@ -414,6 +414,8 @@ $(addprefix $(addprefix $(1), $(notdir $(abspath $(1)))), .json) $(addprefix $(a
 		--node-type-gen-h=$(addprefix sol-flow/, $(addprefix $(notdir $(abspath $(1))), .h)) \
 		$(2)
 	$(eval cleanup-files += $(addprefix $(addprefix $(1), $(notdir $(abspath $(1)))), .json) $(addprefix $(addprefix $(1), $(notdir $(abspath $(1)))), .c))
+$(addprefix $(addprefix $(1), $(notdir $(abspath $(1)))), .c): $(addprefix $(addprefix $(1), $(notdir $(abspath $(1)))), .json)
```

Comment:
it really seems better. that fixed the issue?
no more breakage when using -j ?
If not, go for it